app/modules/individuals/resources.py

```diff
@@ -164,10 +168,52 @@ def post(self, args):
                     code=500,
                 )
 
+        names = []
+        if 'names' in request_in and isinstance(request_in['names'], list):
+            from flask_login import current_user
+            from app.modules.names.models import Name
+
+            for name_json in request_in['names']:
+                preferring_users = []
+                # this is not getting parsed right FIXME
+                if 'preferring_users' in name_json and isinstance(
+                    name_json['preferring_users'], list
+                ):
+                    from app.modules.users.models import User
+
+                    for user_guid in name_json['preferring_users']:
+                        user = User.query.get(user_guid)
+                        if not user:
+                            AuditLog.frontend_fault(
```

Comment:
Would it make more sense if the cleanup_and_abort did the audit log for all cases, at the moment there's a mix in this file and I think it could make debugginng difficult